Formula/arangodb.rb

```diff
@@ -73,7 +67,10 @@ def install
       -DUSE_GOOGLE_TESTS=Off
       -DCMAKE_INSTALL_LOCALSTATEDIR=#{var}
     ]
-    args << "-DTARGET_ARCHITECTURE=nehalem" if build.bottle? && Hardware::CPU.intel?
+    llvm = Formula["llvm@13"] if OS.mac?
+    args << "-DCMAKE_C_COMPILER=#{llvm.opt_bin}/clang" if OS.mac?
+    args << "-DCMAKE_CXX_COMPILER=#{llvm.opt_bin}/clang++" if OS.mac?
```

Comment:
Why can't we use the system clang?

Reply:
@KVS85 if the `clang` version was being pinned here for the same reason the GCC version was, let's try with the host compiler on macOS as well.  The current approach on macOS is bypassing our compiler shim which can be problematic.

Reply:
@danielnachun can you elaborate? We moved to a specific version of llvm in order to control its behaviour and known bugs. All tests (release, pr) of a specific ArangoDB version will be done against this version.

It will compile with the system clang but it will be complete untested.

Reply:
@SMillerDev we only tested against during the last month because we had issues with the system clang. It might now work or not. We are no longer testing the system clang. So the choice is either to use an untested version or llvm. What do you prefer?

Reply:
The system version